### ISSUE TITLE
$ref values must be RFC3986-compliant percent-encoded URIs

### ISSUE THREAD

**Issue by oben01** [oatpp/oatpp-swagger#38]
Hello,
am using swagger editor to generate yaml documentation from /api-docs/oas-3.0.0.json.
Am getting a $ref values must be RFC3986-compliant percent-encoded URIs
Oatpp generate enum with their type between parenthesis
Example from the yaml  : 
$ref: '#/components/schemas/KeywordAccessMode(String)'

My KeywordAccessMode is declared as : 
```c++
#ifndef KeywordAccessModeDTO_hpp
#define KeywordAccessModeDTO_hpp

#include "oatpp/core/macro/codegen.hpp"
#include "oatpp/core/Types.hpp"

#include OATPP_CODEGEN_BEGIN(DTO)

ENUM(KeywordAccessMode, v_int32,
   VALUE(NO_ACCESS, 1, "NoAccess"),
   VALUE(GET, 2, "Get"),
   VALUE(GET_SET, 3, "GetSet")
);

#include OATPP_CODEGEN_END(DTO)

#endif /* KeywordAccessModeDTO_hpp */

```

```yml
openapi: 3.0.0
info:
  title: Yadoms server Swagger-UI
  description: 'Yadoms is open source, simple, powerfull, flexible and multiplatforms domotic solution.'
  contact:
    name: Oussama DAHMAZ
    url: 'http://yadoms.com/forum/'
  license:
    name: GNU General Public License v3.0
    url: 'https://github.com/Yadoms/yadoms-swagger-spot/blob/master/LICENSE'
  version: '1.0'
servers:
  - url: 'http://localhost:8080/rest'
    description: server on localhost
paths:
  '/device/matchcapacitytype/{keywordAccessMode}/{keywordDataType}':
    get:
      description: Get the device list which support a capacity type
      summary: Get devices with capacity Type
      operationId: getDeviceWithCapacityType
      tags:
        - Device
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/EmptyResponse'
      parameters:
        - name: keywordAccessMode
          in: path
          required: true
          deprecated: false
          schema:
            $ref: '#/components/schemas/KeywordAccessMode(String)'
        - name: keywordDataType
          in: path
          required: true
          deprecated: false
          schema:
            $ref: '#/components/schemas/KeywordDataType(String)'
 
components:
  schemas:
    KeywordAccessMode(String):
      type: string
      enum:
        - NoAccess
        - Get
        - GetSet

    KeywordDataType(String):
      type: string
      enum:
        - NoData
        - String
        - Numeric
        - Bool
        - Json
        - Enum
        - DateTime
    EmptyResponse:
      type: object
      properties:
        result:
          type: boolean
        message:
          type: string
        data:
          type: object
          default: {}
          additionalProperties:
            type: string
```

**Comment by lganzzzo** [comment]
I think we can fix that by just replacing `EnumName(Type)` with `EnumName_Type`...

Hey @oben01 please always try to post a minimal representative version of the code, instead of a full version.
There is no chance anyone will find anything in such a huge YAML file.

**Comment by oben01** [comment]
My bad, i just updated my post with a minimal version

**Comment by oben01** [comment]
fixed in [effa2635fc772260f17117bd15e256a44fdbbc18](https://github.com/oatpp/oatpp-swagger/commit/effa2635fc772260f17117bd15e256a44fdbbc18)
